src/core/lib/transition-manager.js

```diff
@@ -11,6 +12,28 @@ export const TRANSITION_EVENTS = {
   IGNORE: 3
 };
 
+const PROP_TYPES = {
+  // transition duration for viewport change
+  transitionDuration: PropTypes.number,
+  // an instance of ViewportTransitionInterpolator, can be used to perform custom transitions.
+  transitionInterpolator: PropTypes.object,
+  // type of interruption of current transition on update.
+  transitionInterruption: PropTypes.number,
+  // easing function
+  transitionEasing: PropTypes.func,
+  // transition status update functions
+  onTransitionStart: PropTypes.func,
+  onTransitionInterrupt: PropTypes.func,
+  onTransitionEnd: PropTypes.func,
+
+  /**
+   * `onViewportChange` callback is fired when the user interacted with the
+   * map. The object passed to the callback contains viewport properties
+   * such as `longitude`, `latitude`, `zoom` etc.
+   */
+  onViewportChange: PropTypes.func
+};
+
 const DEFAULT_PROPS = {
```

Comment:
Thought: Given the extensive props-based API, should this class have a `setProps` method? Many of the other JS classes do. Let's revisit later when we finalize the pure JS API.